core/board.py

```diff
@@ -27,11 +25,21 @@ def update(self, user_shot_choice, is_hit):
                 'I': 8,
                 'J': 9,
                 }
+
+    def update(self, user_shot_choice, is_hit):
         user_letter = user_shot_choice[0]
         user_num = user_shot_choice[1:]
         
-        self.state[rows[user_num]][columns[user_letter]] = 'Hit' if is_hit else 'Miss' 
-            
+        if is_hit == 'Hit':
+            self.state[self.rows[user_num]][self.columns[user_letter]] = 'Hit' 
```

Comment:
also I think I'd probably prefer some setting these to local variables. there's a lot going in on in that line. Maybe `x_coordinate = self.rows[user_num]` `y_coordinate = self.columns[user_letter]`  then this line becomes the much more readable imho `self.state[[x_coordinate][y_coordinate]]`